src/main/java/apleasebot/tasks/TaskList.java

```diff
@@ -0,0 +1,118 @@
+package apleasebot.tasks;
+
+import apleasebot.exceptions.APleaseBotException;
+import apleasebot.exceptions.DataException;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import apleasebot.exceptions.APleaseBotException;
+import apleasebot.exceptions.DataException;
+
+/**
+ * Class that aggregates the different tasks and provide some methods to use
+ */
+public class TaskList {
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Default Constructor for an empty TaskList object
+     */
+    public TaskList() {
+
+    }
+
+    /**
+     * Constructor that will initialise the TaskList object with tasks stored in the storage file
+     * @param list Formatted list that contains stored tasks in String form
+     * @throws DataException Exception thrown if there is an issue
+     */
+    public TaskList(List<String> list) throws DataException {
+        if (list.isEmpty()) {
+            throw new DataException("Data file empty");
+        }
+        int elements = list.size();
+        int i = 0;
+
+        while (i < elements) {
+            String[] str = list.get(i).split(",");
+            switch (str[0]) {
+            case "T":
+                this.add(new Todo(str[2], boolify(str[1])));
+                i++;
+                continue;
+            case "D":
+                this.add(new Deadline(str[2], boolify(str[1]), LocalDateTime.parse(str[3])));
+                i++;
+                continue;
+            case "E":
+                this.add(new Event(str[2], boolify(str[1]), LocalDateTime.parse(str[3]), LocalDateTime.parse(str[4])));
+                i++;
+                continue;
+            default:
+                break;
+            }
+        }
+    }
+
+    public void add(Task task) {
+        this.tasks.add(task);
+    }
+
+    public void remove(int index) {
+        this.tasks.remove(index);
+    }
+
+    public Task get(int index) {
+        return this.tasks.get(index);
+    }
```

Comment:
Perhaps can add "Task" after each method name just to be clear of the method purpose 👍🏻
```suggestion
    public void addTask(Task task) {
        this.tasks.add(task);
    }

    public void removeTask(int index) {
        this.tasks.remove(index);
    }

    public Task getTask(int index) {
        return this.tasks.get(index);
    }
```